### DISCUSSION TITLE
Help creating symlinks

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#546] [Getting started] [answered]
I am working on a formula for my own Tap, but I am struggling with creating symlinks for some files.

This is my scenario, two files have the `.sh` extension, and one does not.

From the docs on [`install_symlink`](https://rubydoc.brew.sh/Pathname.html#install_symlink-instance_method) I do see that I can pass a `Hash`, but I am not well versed in Ruby, and I have not seen something I can use as example on all formulae.

Taking inspiration from [`pyenv.rb`](https://github.com/Homebrew/homebrew-core/blob/99e7dab6a4964d4e648c99ce0d451bf22c93cee2/Formula/pyenv.rb#L41-L43):
```ruby
    %w[pyenv-install pyenv-uninstall python-build].each do |cmd|
      bin.install_symlink "#{prefix}/plugins/python-build/bin/#{cmd}"
    end
```

Taking `pyenv.rb` as inspiration:
```ruby
%w[file_1.sh file_2.sh file_3].each do |cmd|
  bin.install_symlink ...
```

As expected, `file_3` (with no extension) creates the symlink as expected, but I would like to know how to create a symlink without the extension.

Thanks.

**Comment by jonchang** [comment, marked as answer]
You can use the fat arrow here, so if you wanted `foo` to point to `libexec/foo.sh` then you could say:

```ruby
bin.install_symlink "#{libexec}/foo.sh" => "foo"
```

**Comment by cesarcoatl** [comment]
I see!

That created a symlink inside the `bin` folder for the formula which then was symlinked from to `/usr/local/bin`.

Thanks, @jonchang!

**Comment by cesarcoatl** [comment]
For anyone interested, this is my final solution:

```ruby
      %w[foo.sh bar.sh baz].each do |cmd|
        bin.install_symlink "#{prefix}/#{cmd}" => "#{cmd.gsub(".sh", "")}"
      end
```
